src/main/java/com/pingcap/tikv/codec/BytesUtils.java

```diff
@@ -52,11 +52,35 @@ public static void writeBytes(CodecDataOutput cdo, byte[] data) {
         }
     }
 
+    // WriteBytesDesc first encodes bytes using EncodeBytes, then bitwise reverses
+    // encoded value to guarantee the encoded value is in descending order for comparison.
+    public static void writeBytesDesc(CodecDataOutput cdo, byte[] data) {
+        writeBytes(cdo, data);
+        byte[] encodedData = cdo.toBytes();
+        // not sure how to empty cdo
```

Comment:
You might reset underlying byte array stream by expose reset method, which keep the cap but reset position mark.